src/main/java/Parser.java

```diff
@@ -0,0 +1,122 @@
+import java.time.LocalDate;
+
+/**
+ * Handles parsing of user inputs.
+ */
+public class Parser {
+
+    /**
+     * Stores the parsed user inputs and prints response messages for user.
+     * @param userInput User input read by scanner.
+     * @param duke Duke object.
+     * @return Duke response message.
+     */
+    public String parse(String userInput, Duke duke) {
+        String[] strArr = userInput.split(" ");
+        try {
+            check(strArr);
+            if (userInput.equals("bye")) {
+                return duke.getUi().bye();
+                //System.exit(0);
+            } else if (userInput.equals("list")) {
+                return duke.getUi().printTaskList(duke.getTasks());
+            } else if (strArr[0].equals("done")) {
+                duke.getTasks().getTask(Integer.parseInt(strArr[1]) - 1).markAsDone();
+                duke.getStorage().writeFile(duke.getTasks().getTaskList());
+                return duke.getUi().printDoneTask(duke.getTasks().getTask(Integer.parseInt(strArr[1]) - 1));
+            } else if (strArr[0].equals("delete")) {
+                int i = Integer.parseInt(strArr[1]) - 1;
+                Task t = duke.getTasks().getTask(Integer.parseInt(strArr[1]) - 1);
+                duke.getTasks().deleteTask(i);
+                duke.getStorage().writeFile(duke.getTasks().getTaskList());
+                return duke.getUi().printDeleteTask(t, duke.getTasks());
+            } else {
+                if (strArr[0].equals("todo")) {
+                    String sd = "";
```

Comment:
Perhaps you can use a more descriptive variable name :) I notice this in other places too in this method

Reply:
Alright!